msal/src/main/java/com/microsoft/identity/client/NativeAuthPublicClientApplication.kt

```diff
@@ -86,6 +96,73 @@ class NativeAuthPublicClientApplication(
         //  To avoid duplicating the code, callback methods are routed through their
         //  coroutine-equivalent through this CoroutineScope.
         val pcaScope = CoroutineScope(SupervisorJob() + Dispatchers.IO)
+
+        fun getCurrentAccountInternal(config: NativeAuthPublicClientApplicationConfiguration): IAccount? {
+            LogSession.logMethodCall(TAG, "${TAG}.getCurrentAccountInternal")
+
+            val params = CommandParametersAdapter.createCommandParameters(
+                config,
+                config.oAuth2TokenCache
+            )
+
+            val command = GetCurrentAccountCommand(
+                params,
+                LocalMSALController(),
+                object : CommandCallback<List<ICacheRecord?>?, BaseException?> {
+                    override fun onTaskCompleted(result: List<ICacheRecord?>?) {
+                        // Do nothing, handled by CommandDispatcher.submitSilentReturningFuture()
+                    }
+
+                    override fun onError(error: BaseException?) {
+                        // Do nothing, handled by CommandDispatcher.submitSilentReturningFuture()
+                    }
+
+                    override fun onCancel() {
+                        // Not required
+                    }
+                },
+                PublicApiId.NATIVE_AUTH_GET_ACCOUNT
+            )
+            val result = CommandDispatcher.submitSilentReturningFuture(command)
+                .get().result as List<ICacheRecord?>?
+
+            // To simplify the logic, if more than one account is returned, the first account will be picked.
+            // We do not support switching from MULTIPLE to SINGLE.
+            return getAccountFromICacheRecordsList(result)
+        }
+
+        /**
+         * Get an IAccount from a list of ICacheRecord.
+         *
+         * @param cacheRecords list of cache record that belongs to an account.
+         * If the list can be converted to multiple accounts, only the first one will be returned.
+         */
+        private fun getAccountFromICacheRecordsList(cacheRecords: List<ICacheRecord?>?): IAccount? {
+            LogSession.logMethodCall(TAG, "${TAG}.getAccountFromICacheRecordsList")
+            if (cacheRecords.isNullOrEmpty()) {
+                return null
+            }
+            val account = AccountAdapter.adapt(cacheRecords)
```

Comment:
Sounds like this is an `accountList` as opposed to an `account`, right? So let's name it appropriately

Reply:
Done